modules/machineset-vsphere-data-disks.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-vsphere.adoc
+// * machine_management/control_plane_machine_management/cpmso_provider_configurations/cpmso-config-options-vsphere.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="machineset-vsphere-data-disks_{context}"]
+= Configuring data disks by using machine sets
+
+{product-title} clusters on {vmw-first} support adding up to 29 disks to the virtual machine (VM) controller.
+
+:FeatureName: Configuring {vmw-short} data disks
+include::snippets/technology-preview.adoc[]
+
+By configuring data disks, you can attach disks to VMs and use them to store data for etcd, container images, and other uses.
+Separating data can help avoid filling the primary disk so that important activities such as upgrades have the resources that they require.
+
+[NOTE]
+====
+Adding data disks attaches them to the VM and mounts them to the location that {op-system} designates.
+To mount the data disks to a specific location, you must configure each machine to use the data disks according to your needs.
+====
+
+.Prerequisites
+
+* You have administrator access to {oc-first} for an {product-title} cluster on {vmw-short}.
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following lines under the `providerSpec` field:
++
+[source,yaml]
+----
+tag::compute[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+end::compute[]
+tag::controlplane[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+end::controlplane[]
+# ...
+spec:
+  template:
+tag::compute[]
+    spec:
+      providerSpec:
+        value:
+          dataDisk: <1>
+          - name: "<disk_name>" <2>
+            provisioningMode: "<mode>" <3>
+            sizeGiB: 20 <4>
+          - name: "<disk_name>"
+            provisioningMode: "<mode>"
+            sizeGiB: 20
+end::compute[]
+tag::controlplane[]
+    machines_v1beta1_machine_openshift_io:
+      spec:
+        providerSpec:
+          value:
+            dataDisk: <1>
+            - name: "<disk_name>" <2>
+              provisioningMode: "<mode>" <3>
+              sizeGiB: 20 <4>
+            - name: "<disk_name>"
+              provisioningMode: "<mode>"
+              sizeGiB: 20
+end::controlplane[]
+# ...
+----
+<1> Specify a collection of 1-29 data disk definitions.
+This sample configuration shows the formatting to include two data disk definitions.
+<2> Specify the name of the data disk.
+The maximum length is 80 characters.
+#Any other name restrictions?#
+<3> Specify the data disk provisioning method.
+This value defaults to the vSphere default storage policy if not set. 
+Valid values are `Thin`, `Thick`, and `EagerlyZeroed`.
+#Do we need to define these, or is this pretty common knowledge for a vSphere admin?#
```

Comment:
Couple questions in the `#` delimited lines here

Reply:
> Any other name restrictions?

Name must consist only of alphanumeric characters, hyphens and underscores, and must start and end with an alphanumeric character: https://github.com/openshift/api/blob/80f8dcff5d6b0311cb448e9d690314cb3102af3a/machine/v1beta1/types_vsphereprovider.go#L194

> Do we need to define these, or is this pretty common knowledge for a vSphere admin?

I normally like to specify it because someone will always and ask what are the valid values.  The installer should correct the user if they try to install w/ invalid values.  Also during Day 2, machinesets will complain if invalid values too.  Also, upstream CAPV documents these fields for other usages so its a common pattern.

Reply:
oh, yeah I definitely agree we need to list the valid disk provisioning method valuess. I was wondering whether we need to define what each valid value means, or if the user will already know what e.g. `Thin` means.